src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -443,6 +445,23 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
                 return !is_type<ov::op::v0::MatMul>(next_node);
             });
 
+#ifdef ENABLE_ONEDNN_FOR_GPU
+        // moe expert requires onednn enabled which needs in order queue.
+        if (config.get_queue_type() == QueueTypes::in_order) {
```

Comment:
I think it would be better to use "supports_immad" instead of in_order queue when your intention is to limit the use case only for "onednn gemm" being used. Because in_order_queue condition can be used for debugging purpose too. For example we use in order queue type to measure layer wise performance in MTL where onednn is not used. In that case, moe might be fused even though onednn is not available in MTL.